sim/dex-data.ts

```diff
@@ -101,15 +115,18 @@ export class BasicEffect implements EffectData {
 	/** ??? */
 	sourceEffect: string;
 
-	constructor(data: AnyObject) {
-		this.exists = true;
-		Object.assign(this, data);
-
+	/**
+	 * Pass 'false' for the second parameter if you only want the declared fields of BasicEffect
+	 * to be initialized - the other properties of data will ignored.
+	 * This is to help w/ V8 hidden classes (want to init fields in consistent order)
+	 */
+	constructor(data: AnyObject, copyOtherFields = true) {
```

Comment:
It looks as if `BasicEffect` should be an abstract class, so that it will never have to worry about copying other fields itself.

Do you happen to know how many other fields there actually are?

Reply:
(I will get back to you on this, I'm just working on making the test suite run faster)

Reply:
Yeah, abstract class makes sense here and would simplify it.

Documented # of other fields in later comment in PR.